types/knockout.mapping/index.d.ts

```diff
@@ -15,13 +15,23 @@ export = self;
 
 
 declare global {
-    // the idea behind this is good, but doesn't work as intended. This will make object properties T | Observable<T>, 
-    //and you will have to put the correct Type in each property to use them, what defeats the purpose. Besides, this gives  
-    //RangeError: Maximum call stack size exceeded in TSC when used in all the mapping signatures. Maybe this can be used with TS 2.8 and conditional typing
+
+    type Primitives = string | number | boolean;
+
+    type MappedType<T> =
+        T extends Primitives ? KnockoutObservable<T> :
+        T extends object ? KnockoutObservableType<T> :
+        any;
+
     type KnockoutObservableType<T> = {
-        [P in keyof T]: KnockoutObservable<KnockoutObservableType<T[P]>> | T[P];
+        [P in keyof T]: T[P] extends Primitives ? KnockoutObservable<T[P]> :
+                        T[P] extends any[] ? KnockoutObservableArrayType<T[P][number]> :
```

Comment:
What happens here if `T[P]` is a `ReadonlyArray<any>` ?

Reply:
Added support for this scenario.